src/main/scala/edu/berkeley/cs/rise/opaque/execution/operators.scala

```diff
@@ -234,42 +234,73 @@ case class EncryptedFilterExec(condition: Expression, child: SparkPlan)
 case class EncryptedAggregateExec(
   groupingExpressions: Seq[NamedExpression],
   aggExpressions: Seq[AggregateExpression],
-  mode: AggregateMode,
+  // Specify mode if ALL aggExpressions should be cast to the provided AggregateMode
+  mode: Option[AggregateMode],
```

Comment:
Seems like mode isn't really needed if this information is embedded in `aggExpressions`?